runelite-client/src/main/java/net/runelite/client/plugins/itemskeptondeath/ItemsKeptOnDeathPlugin.java

```diff
@@ -223,73 +232,92 @@ private void rebuildItemsKeptOnDeathInterface()
 		final ItemContainer equipment = client.getItemContainer(InventoryID.EQUIPMENT);
 		final Item[] equip = equipment == null ? new Item[0] : equipment.getItems();
 
+		final List<List<ItemStack>> itemLists = calculateKeptLostItems(inv, equip);
+
+		final List<Widget> keptItems = itemLists.get(0).stream()
+			.map(item -> createItemWidget(kept, item, true)).collect(Collectors.toList());
+		final List<Widget> lostItems = itemLists.get(1).stream()
+			.map(item -> createItemWidget(lost, item, false)).collect(Collectors.toList());
+
+		int rows = (keptItems.size() + MAX_ROW_ITEMS - 1) / MAX_ROW_ITEMS;
+		// Show an empty row if there isn't anything
+		if (rows > 0)
+		{
+			// ORIGINAL_LOST_Y/HEIGHT includes a row already
+			rows--;
+		}
+		// Adjust items lost container position if new rows were added to kept items container
+		lost.setOriginalY(ORIGINAL_LOST_Y + (rows * ITEM_Y_STRIDE));
+		lost.setOriginalHeight(ORIGINAL_LOST_HEIGHT - (rows * ITEM_Y_STRIDE));
+		positionWidgetItems(kept, keptItems);
+		positionWidgetItems(lost, lostItems);
+
+		updateKeptWidgetInfoText(keptItems, lostItems);
+	}
+
+	/**
+	 * Calculates which items will be kept/lost. first list is kept items, second is lost.
+	 * @param inv players inventory
+	 * @param equip players equipement
+	 * @return list of items kept followed by a list of items lost
+	 */
+	@VisibleForTesting
+	List<List<ItemStack>> calculateKeptLostItems(final Item[] inv, final Item[] equip)
```

Comment:
I'd much rather this return a static inner class

Reply:
You could also then move back `hasAlwaysLost` then too probably